doc/classes/WorkerThreadPool.xml

```diff
@@ -1,8 +1,72 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <class name="WorkerThreadPool" inherits="Object" version="4.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
 	<brief_description>
+		A singleton that executes [Callable]s in a multithreaded way.
 	</brief_description>
 	<description>
+		Just like the signal system, the [WorkerThreadPool] uses [Callable]s. The [WorkerThreadPool] executes them in a multithreaded way. You can schedule the worker to process one [Callable] as an individual task. Which is then executed in another thread. Or schedule multiple elements of one [Callable] in a group task. These are executed in one or more threads. You decide. Set a task as [b]high_priority[/b] to let the worker process them earlier than others.
+		To schedule [Callable]s, pass them on to [method add_task] and [method add_group_task]:
+		[codeblocks]
+		[gdscript]
+		func do_something() -&gt; void:
+		    var is_very_important : bool = true
+		    var task_id : int = WorkerThreadPool.add_task(self.load_cheese_cake, is_very_important, "do it fast, but elsewhere")
+		    # and now we wait…
+		    WorkerThreadPool.wait_for_task_completion(task_id)
+
+		    # Now together
+		    # But only if this class inherits from [Object].
+		    WorkerThreadPool.add_group_task(self._load_file_threaded, to_load.size(), -1, !is_very_important, "load the cheese, take your time")
+
+		func load_cheese_cake() -&gt; void:
+		    # total secret where it comes from
+
+
+		var to_load : Array = ["ham.png", "cheese.png", "baguette.png"]
+
+
+		func _load_file_threaded(index : int) -&gt; void:
+		    index = min(index, to_load.size() - 1)
+		    index = max(0, index)
+		    load_file(to_load[index])
+
+
+		func load_file(path : String) -&gt; void:
+		    # code to load file comes here
+		[/gdscript]
+		[csharp]
+		public void DoSomething()
+		{
+		    bool is_very_important = true;
+		    int task_id = WorkerThreadPool.add_task(this.LoadCheeseCake, is_very_important, "do it fast, but elsewhere");
+		    // and now we wait…
+		    WorkerThreadPool.wait_for_task_completion(task_id);
+
+		    // Now together.
+		    // But only if this class inherits from [Object].
+		    WorkerThreadPool.add_group_task(this._LoadFileThreaded, to_load.Length, -1, !is_very_important, "load the cheese, take your time");
+		}
+
+		public void LoadCheeseCake()
+		{
+		    // total secret where it comes from
+		}
+
+		private Godot.Collections.Array&lt;String&gt; to_load = new Godot.Collections.Array&lt;String&gt;{"ham.png", "cheese.png", "baguette.png"};
+
+		public void _LoadFileThreaded(int index)
```

Comment:
By C# naming conventions methods should not be prefixed by an underscore `_`, I believe this is the convention used in GDScript for "private" members (since private doesn't exist) so it's likely that the best equivalent would be for this method to be declared as private:

```suggestion
		private void LoadFileThreaded(int index)
```

Also, the parameter might need to be a `long` but I'm not sure about this, in C++ this parameter is the `work_index` declared as a `uint32_t` (in C# that would be `uint`) but since it gets sent as a Variant it would most likely arrive as a `Variant::INT` which is a 64-bit integer (a `long` in C#).

Reply:
Hm… Difficult.  
Judging from the Variant.cs file (just skimming over): If it gets a C# `long`, it turns it into an `int` and returns a C# `long` only if gets an `int64`. a C++ `uint32` would turn into a C# `uint`. So I'll go for `uint` for now

Reply:
updated PR description, so I'll mark this as "resolved"